src/utils/firstSentenceFromContent.ts

```diff
@@ -0,0 +1,16 @@
+import { StrapiPageContentBlock, TextBlock } from "../types/strapi"
+
+const firstSentenceFromContent = (content: StrapiPageContentBlock[]) => {
+  const textBlock = content.find(
+    (block) => block.strapi_component === "common-content.text-block"
+  ) as TextBlock | undefined
+  const firstTextBlock = textBlock?.text.data.text
+  const firstSentence = firstTextBlock
+    ?.replace(/#+.+\n/g, "") // Remove headings
+    ?.split(/[.\n]/) // Split by sentence or line break
+    .find((string) => string && string.length > 0) // Find first non-empty string
+    ?.concat(".")
+  return firstSentence
+}
```

Comment:
This keeps Markdown markup, such as links, images or styling.

I think we're better off omitting `description` on content pages, or using a Markdown parser if we insist to have one autogenerated.

Also, it might be better to get a full paragraph - whatever is displaying the `description` is usually smart enough to clip it itself.